lib/deterministicmnlist/SimplifiedMNList.js

```diff
@@ -54,18 +65,37 @@ SimplifiedMNList.prototype.applyDiff = function applyDiff(simplifiedMNListDiff)
   this.addOrUpdateMNs(diff.mnList);
 
   this.lastDiffMerkleRootMNList = diff.merkleRootMNList || constants.NULL_HASH;
+
   this.merkleRootMNList = this.calculateMerkleRoot();
 
-  if (!this.verify()) {
+  if (this.lastDiffMerkleRootMNList !== this.merkleRootMNList) {
     throw new Error("Merkle root from the diff doesn't match calculated merkle root after diff is applied");
   }
 
   this.cbTx = new Transaction(diff.cbTx);
   this.cbTxMerkleTree = diff.cbTxMerkleTree.copy();
 
+  this.quorumsActive = this.cbTx.version >= 2;
+
+  if (this.quorumsActive) {
+    this.deleteQuorums(diff.deletedQuorums);
+    this.addAndMaybeRemoveQuorums(diff.newQuorums);
+    this.lastDiffMerkleRootQuorums = diff.merkleRootQuorums || constants.NULL_HASH;
+    this.merkleRootQuorums = this.calculateMerkleRootQuorums();
+
+    if (this.lastDiffMerkleRootQuorums !== this.merkleRootQuorums) {
+      throw new Error("Quorum merkle root from the diff doesn't match calculated merkle root after diff is applied");
+    }
+  }
+
   this.validMNs = this.mnList.filter(function (smlEntry) {
     return smlEntry.isValid;
   });
+  /*
```

Comment:
Commented by mistake?

Reply:
Yep, forgot to remove. Will probably come back in the 2nd PR in one way or other.